### PR TITLE
Check the resource bundle is around before adding it to native image

### LINKED ISSUE #21694: Hibernate Validator warnings are displayed when optional ResourceBundles are missing
### Describe the bug

When an app depending on the Hibernate Validator extension is package using the native build, the following messages are printed to the output:

``` 
The bundle named: ContributorValidationMessages, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
The bundle named: ValidationMessages, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
The bundle named: messages, has not been found. If the bundle is part of a module, verify the bundle name is a fully qualified class name. Otherwise verify the bundle path is accessible in the classpath.
```

Although that does no harm to the build, it may lead to confusion giving that those bundles are optional.

### Expected behavior

Do not print any message if the bundle does not exist in the user's application

### Actual behavior

The message is printed.

### How to Reproduce?

1. Clone https://github.com/quarkusio/registry.quarkus.io
2. run `mvn clean package -Dnative -DskipTests` 

### Output of `uname -a` or `ver`

Fedora 35

### Output of `java -version`

11

### GraalVM version (if different from Java)

21

### Quarkus version or git rev

2.5.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.4

### Additional information

This issue was originated from [a chat in Zulip](https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Hibernate.20Validation). 

Also it seems that [this snippet](https://github.com/quarkusio/quarkus/blob/main/extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java#L346-L353) may need to be enhanced to check if the bundle really exists

### PR DESCRIPTION
**Pull request by gsmet**
Fixes #21694

### CODE REVIEW

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java [hunk 1]
```diff
@@ -345,11 +345,28 @@ public void build(HibernateValidatorRecorder recorder, RecorderContext recorderC
...
     NativeImageConfigBuildItem nativeImageConfig() {
-        return NativeImageConfigBuildItem.builder()
-                .addResourceBundle(AbstractMessageInterpolator.DEFAULT_VALIDATION_MESSAGES)
-                .addResourceBundle(AbstractMessageInterpolator.USER_VALIDATION_MESSAGES)
-                .addResourceBundle(AbstractMessageInterpolator.CONTRIBUTOR_VALIDATION_MESSAGES)
-                .build();
+        List<String> potentialHibernateValidatorResourceBundles = List.of(
+                AbstractMessageInterpolator.DEFAULT_VALIDATION_MESSAGES,
+                AbstractMessageInterpolator.USER_VALIDATION_MESSAGES,
+                AbstractMessageInterpolator.CONTRIBUTOR_VALIDATION_MESSAGES);
```
Comment (gsmet) [inline extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java:351]: I kept the list here as there's really no need to keep it static if we are not building a native image.

Comment (gastaldi) [reply extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java:351]: That works. You can also use `String[]` and save some memory if you're only iterating.

Comment (gastaldi) [inline extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java:351]:
```suggestion
        String[] potentialHibernateValidatorResourceBundles = {
                AbstractMessageInterpolator.DEFAULT_VALIDATION_MESSAGES,
                AbstractMessageInterpolator.USER_VALIDATION_MESSAGES,
                AbstractMessageInterpolator.CONTRIBUTOR_VALIDATION_MESSAGES
                };
```

**Review by gastaldi — APPROVED** [review]
LGTM, added a minor comment. Feel free to ignore it if it doesn't make much sense

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java [hunk 2]
```diff
@@ -345,11 +345,28 @@ public void build(HibernateValidatorRecorder recorder, RecorderContext recorderC
...
+        List<String> userDefinedHibernateValidatorResourceBundles = new ArrayList<>();
```
Comment (gastaldi) [inline extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java:352]: Minor nitpick: you can size the list with `potentialHibernateValidatorResourceBundles.length`

**Comment by gsmet** [comment]
Yeah, it's really not a hot spot so I don't think it's worth the hassle. Thanks for the comments though.

**Pull request merged**